sandbox/libs/authn/src/main/java/org/opensearch/authn/AuthenticationTokenHandler.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.authn;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.shiro.authc.AuthenticationToken;
+import org.apache.shiro.authc.UsernamePasswordToken;
+import org.opensearch.authn.tokens.BasicAuthToken;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * Extracts Shiro's {@link AuthenticationToken} from different types of auth headers
+ *
+ * @opensearch.experimental
+ */
+public class AuthenticationTokenHandler {
+
+    private static final Logger logger = LogManager.getLogger(AuthenticationTokenHandler.class);
+
+    /**
+     * Extracts shiro auth token from the given header token
+     * @param authenticationToken the token from which to extract
+     * @return the extracted shiro auth token to be used to perform login
+     */
+    public static AuthenticationToken extractShiroAuthToken(org.opensearch.authn.tokens.AuthenticationToken authenticationToken) {
+        AuthenticationToken authToken = null;
+
+        if (authenticationToken instanceof BasicAuthToken) {
+            authToken = handleBasicAuth((BasicAuthToken) authenticationToken);
+        }
+        // TODO: check for other type of HeaderTokens
+        return authToken;
+    }
+
+    /**
+     * Returns auth token extracted from basic auth header
+     * @param token the basic auth token
+     * @return the extracted auth token
+     */
+    private static AuthenticationToken handleBasicAuth(final BasicAuthToken token) {
+
+        final byte[] decodedAuthHeader = Base64.getDecoder().decode(token.getHeaderValue().substring("Basic".length()).trim());
+        String decodedHeader = new String(decodedAuthHeader, StandardCharsets.UTF_8);
+
+        final String[] decodedUserNamePassword = decodedHeader.split(":");
+
+        // Malformed AuthHeader strings
+        if (decodedUserNamePassword.length != 2) return null;
+
+        logger.info("Logging in as: " + decodedUserNamePassword[0]);
```

Comment:
`:` is accepted in the password according to this comment in the security plugin codebase: https://github.com/opensearch-project/security/blob/cad7c7201e902aeb711b407648d41d3c55fe7a49/src/main/java/org/opensearch/security/support/HTTPHelper.java#L52-L57

Reply:
What do you think about creating an issue and finding/implementing a better version of this?

🤞 I would rather pull a small library in for this kind of logic

Reply:
I will modify the logic to be similar to one that Craig posted above. However, this seems like a good avenue to explore libraries that provide support for this.